tests/src/Unit/GroupManagerTest.php

```diff
@@ -332,35 +340,41 @@ protected function addNewDefaultRole($entity_type, $bundle, $role_name) {
 
     // It is expected that the OG permissions that need to be populated on the
     // new role will be requested from the PermissionEvent listener. In order to
     // get the role name, this will be requested from the OgRole object.
-    $this->eventDispatcherProphecy->dispatch(PermissionEventInterface::EVENT_NAME, Argument::type('\Drupal\og\Event\PermissionEvent'))
-      ->willReturn($this->permissionEventProphecy->reveal())
-      ->shouldBeCalled();
+//    $this->eventDispatcherProphecy->dispatch(PermissionEventInterface::EVENT_NAME, Argument::type('\Drupal\og\Event\PermissionEvent'))
+//      ->willReturn($this->permissionEventProphecy->reveal())
+//      ->shouldBeCalled();
 
     // It is expected that the role will be created with default properties.
-    $this->entityStorageProphecy->create(OgRole::getDefaultRoleProperties($role_name))
-      ->will(function () use ($entity_type, $bundle, $role_name, $permission_event, $og_role) {
-        // For each role that is created it is expected that the role name will
-        // be retrieved, so that the role name can be used to filter the
-        // permissions.
-        // This type of behavior is mocked in Prophecy using a 'promise' - the
-        // call to getName() returns a different result depending on the last
-        // call that was made to EntityStorageInterface::create(), and by itself
-        // it changes the argument that is used for filterByDefaultRole().
-        // @see https://github.com/phpspec/prophecy#arguments-wildcarding
+    $this->entityStorageProphecy->create($this->getDefaultRoleProperties($role_name))
+      ->will(function () use ($entity_type, $bundle, $role_name, $og_role, $permission_manager) {
+        $permission_manager->getDefaultPermissions($entity_type, $bundle, [], $role_name)
+          ->willReturn([])
+          ->shouldBeCalled();
+//        // For each role that is created it is expected that the role name will
+//        // be retrieved, so that the role name can be used to filter the
+//        // permissions.
+//        // This type of behavior is mocked in Prophecy using a 'promise' - the
+//        // call to getName() returns a different result depending on the last
+//        // call that was made to EntityStorageInterface::create(), and by itself
+//        // it changes the argument that is used for filterByDefaultRole().
+//        // @see https://github.com/phpspec/prophecy#arguments-wildcarding
         $og_role->getName()
-          ->will(function () use ($role_name, $permission_event) {
-            $permission_event->filterByDefaultRole($role_name)
-              ->willReturn([])
-              ->shouldBeCalled();
-            return $role_name;
-          })
+          ->willReturn($role_name)
+//            $permission_event->filterByDefaultRole($role_name)
+//              ->willReturn([])
+//              ->shouldBeCalled();
+//            return $role_name;
+//          })
```

Comment:
Sorry for the mess. It took a bit more time than expected to get the tests green yesterday evening and I had to rush out to catch my train before I was ready with the clean up.

I'll clean this up now and rebase the branch so this unholy mess disappears.